File: BackendDBD/src/main/java/uni/edu/pe/modulo_crm/dto/Mantenimientodto/Infraestructura.java
package uni.edu.pe.modulo_crm.dto.Mantenimientodto;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Infraestructura {
    private String idInfraestructura;
    private String nombre;
    private Date fechaAdquisicion;
    private String ubicacion;
    private String estado;
    private String tipo;
    private String responsable;
    private Integer frecuenciaMantenimiento;
    private Date ultimoMantenimiento;
    private Date proximoMantenimiento;
}

File: FrontendDBD/frontend/src/app/models/modelsmantenimiento/infraestructura.ts
import { Empleado } from "../empleado.model";

export interface Infraestructura {
    idInfraestructura: string;
    nombre: string;
    fechaAdquisicion: Date | null;
    ubicacion: string;
    estado: string;
    tipo: string;
    responsable: string;
    frecuenciaMantenimiento: number;
    ultimoMantenimiento: Date | null;
    proximoMantenimiento: Date | null;
}